lib/appendGitlabProv.js

```diff
@@ -0,0 +1,72 @@
+import rdf from 'rdf-ext'
+import { Transform } from 'readable-stream'
+import { checkGitlabVars, provFromGitlab } from './metadata/produceProv.js'
+import { prov } from './namespaces.js'
+
+const typePredicate = rdf.namedNode(
+  'http://www.w3.org/1999/02/22-rdf-syntax-ns#type')
+
+class ProvMetadata extends Transform {
+  constructor (context, { subjectsWithClass, graph }) {
+    super({ objectMode: true })
+    this.type = subjectsWithClass
+    const { omitProv, message } = checkGitlabVars()
+    if (omitProv) {
+      console.warn(message)
+      this.omitProv = omitProv
+    } else {
+      this.provPointer = provFromGitlab()
+    }
+    this.graph = graph
+  }
+
+  _transform (quad, encoding, callback) {
+    if (quad.predicate.equals(typePredicate) && quad.object.equals(this.type) &&
+      !this.omitProv) {
+      this.provPointer.addOut(prov.generates, quad.subject)
```

Comment:
You mean https://prefix.zazuko.com/prov:generated?
```suggestion
      this.provPointer.addOut(prov.generated, quad.subject)
```